modules/fraud_detection/fraud_detection.c

```diff
@@ -278,7 +278,7 @@ static int check_fraud(struct sip_msg *msg, str *user, str *number, int *pid)
 
 	/* Find a rule */
 
-	unsigned int matched_len;
+	unsigned int matched_len = 0;
 	lock_start_read(frd_data_lock);
 	rt_info_t *rule = drb.match_number(*dr_head, *pid, number, &matched_len);
```

Comment:
the `match_number` function does not populate `matched_len` if error (no match) is returned. This is correct. But in this case `rule` will be NULL and the `check_fraud` will exit (see bellow). 
So I don't see the case where `matched_len` may be used in-init'ed.

Reply:
`match_number` call `find_rule_by_prefix_unsafe`.
in `find_rule_by_prefix_unsafe`, if `get_prefix` got NULL, it will try `check_rt` for prefixless rules again.
`matched_len` is not set by the branch. and we got `rule` in the branch.